packages/runtime/container-runtime/src/batchManager.ts

```diff
@@ -50,11 +56,15 @@ export class BatchManager {
         // If we were provided soft limit, check for exceeding it.
         // But only if we have any ops, as the intention here is to flush existing ops (on exceeding this limit)
         // and start over. That's not an option if we have no ops.
-        if (this.options.softLimit !== undefined && this.length > 0 && socketMessageSize >= this.options.softLimit) {
+        if (this.options.softLimit !== undefined
+            && this.length > 0
+            && socketMessageSize >= this.options.softLimit
+            && socketMessageSize < (this.options.compressionOptions?.minimumBatchSizeInBytes ?? 0)) {
```

Comment:
This will never evaluate to true if there's no `compressionOptions`. I think maybe the check should be regardless of the `socketMessageSize`. ie just see if `minimumBatchSizeInBytes ?? infinity` is infinity.

Reply:
Realized this after pushing it and changed the default case to Infinity instead of 0 after realizing my mistake. Would we rather check `Infinity === minimumBatchSizeInBytes ?? Infinity` or as I have it (`socketMessageSize < minimumBatchSizeInBytes ?? Infinity`)?

Reply:
I'd prefer `Infinity === minimumBatchSizeInBytes ?? Infinity` as the other one adds some states related to the relationship between the soft/hard limits and the `minimumBatchSizeInBytes` value which I'm not certain are (or should be) covered by unit tests right now. I think it's fine to just keep the existing behavior if the feature is disabled.

If we're going down the path of infinite batches, the batchmanager separation between attach messages and regular messages will become obsolete, so this logic will likely be going away or at least be refactored.